netket/operator/_kinetic.py

```diff
@@ -0,0 +1,71 @@
+from typing import Optional, Union
+
+from netket.utils.types import DType, PyTree
+
+from netket.hilbert import AbstractHilbert
+from netket.operator import ContinousOperator
+
+import jax
+import jax.numpy as jnp
+
+
+class KineticEnergy(ContinousOperator):
+    def __init__(
+        self,
+        hilbert: AbstractHilbert,
+        mass: Union[float, list],
+        dtype: Optional[DType] = float,
+    ):
+        r"""Args:
+        mass: float if all masses are the same, list indicating the mass of each particle otherwise
+        """
+
+        self._dtype = dtype
+        self._mass = mass
```

Comment:
please write the formula for the kinetic energy, also specifying that you are implicitly taking hbar=1

Reply:
store it as a jax array
```suggestion
        if not hasattr(mass, "__len__"):
            mass = [mass for _ in range(hilbert. num particles?)]
        self._mass = jnp.asarray(mass, dtype=dtype)
```